powersimdata/input/transform_profile.py

```diff
@@ -1,111 +1,98 @@
         self.scenario_info = scenario_info
-        self.grid = copy.deepcopy(grid)
+
         self.ct = copy.deepcopy(ct)
+        self.grid = copy.deepcopy(grid)
+
         self.scale_keys = {
-            "wind": {"wind", "wind_offshore"},
-            "solar": {"solar"},
-            "hydro": {"hydro"},
-            "demand": {"demand"},
+            "wind": ["wind", "wind_offshore"],
+            "solar": ["solar"],
+            "hydro": ["hydro"],
+            "demand": ["demand"],
         }
-        self.n_new_plant = (
-            0 if "new_plant" not in self.ct.keys() else len(self.ct["new_plant"])
-        )
-        self.base_plant = Grid(self.grid.interconnect).plant
+        self.n_new_plant, self.n_new_clean_plant = self._get_number_of_new_plant()
+
+    def _get_number_of_new_plant(self):
+        """Return the total number of new plant and new clean plant.
+
+        :return: (*tuple*) -- first element is the total number of new plant and second
+            element is the total number of new clean plant (*hydro*, *solar*,
+            *onshore wind* and *offshore wind*).
+        """
+        n_plant = [0, 0]
+        if "new_plant" in self.ct.keys():
+            for p in self.ct["new_plant"]:
+                n_plant[0] += 1
+                if p["type"] in sum(self.scale_keys.values(), []):
+                    n_plant[1] += 1
+        return n_plant
 
     def _get_renewable_profile(self, resource):
-        """Returns the transformed grid.
+        """Return the transformed profile.
 
         :param str resource: *'hydro'*, *'solar'* or *'wind'*.
-        :return: (*pandas.DataFrame*) -- power output for generators of
-            specified type with plant identification number  as columns and
-            UTC timestamp as index.
+        :return: (*pandas.DataFrame*) -- power output for generators of specified type
+            with plant identification number as columns and UTC timestamp as indices.
         """
-        power_output = self._input_data.get_data(self.scenario_info, resource)
-        if not bool(self.ct):
-            return power_output
+        plant_id = (
+            self.grid.plant.iloc[: len(self.grid.plant) - self.n_new_plant]
+            .groupby("type")
+            .get_group(resource)
+            .index
+        )
+        profile = self._input_data.get_data(self.scenario_info, resource)[plant_id]
+        scaled_profile = profile * self.grid.plant.loc[plant_id, "Pmax"]
+
+        if self.n_new_clean_plant > 0:
+            new_profile = self._add_plant_profile(profile, resource)
+            return scaled_profile.join(new_profile)
         else:
-            if self.n_new_plant > 0:
-                power_output = self._add_plant_profile(power_output, resource)
-            if resource in self.ct.keys():
-                power_output = self._scale_plant_profile(power_output, resource)
-            return power_output
+            return scaled_profile
```

Comment:
We are not dealing with `wind_offshore` here, are we?

Reply:
Ah. The `plant_id` list is wrong for wind. I need to include `wind_offsore`. Thanks!

Reply:
Fixed. Tests have been refactored accordingly.